### ISSUE TITLE
Image from URL (Alexa)

### ISSUE THREAD

**Issue by AlexTeusz** [gitbrent/PptxGenJS#318]
Hi, 

I want to use an image from AWS S3 for my PPTX but there is no option to use an url in the method 
`pptx.addImage()
`
Has anyone an idea how to fix this?
Because I'm using the AWS Lambda for my Alexa Skill, I can't store the images in an existing folder. 

Thank you :) 

Alex

**Comment by gitbrent** [comment]
Hi @AlexTeusz,

Thanks for the suggestion.  This would be a Node-only feature once implemented as only local-server images are supported in browsers.

A work-around would be fetching the image yourself, encoding it into base64, then passing it as the `data` value, or creating an `<img>` and populating it using "src=amazon.c3.url", then passing its ID to `addImage()`.

**Comment by DSheffield** [comment]
I needed to do a similar thing and adapted approach 1 in this answer this as a starting point : 
https://stackoverflow.com/a/20285053

**Comment by gitbrent** [comment]
Nah, that was back when i was using an `<img>` tag to get the data from.

Check the [docs](https://gitbrent.github.io/PptxGenJS/docs/api-images.html) for newest options.

**Comment by nick-vfip** [comment]
I'd suggest having the feature back, as it is more convenient to add images already present in the DOM than to pass on URLs (dynamically produced) and avoid having issues with the CORS.

It happened in my case. I was trying to load an image from AWS which is different for each logged-in user, into the master slide, and got the CORS issue related to AWS. I tried to convert it to the data format, but still, the CORS issue persisted. Then I finally settled to download the image dynamically and save it in my public folder, give it a common name, and used that name to get my image into the master slide. It'd been easier if the image tag feature was there.
